### PR TITLE
[KernelGen][Nvidia] Add acos_ operator with Triton kernel

### PR DESCRIPTION
**Pull request by XDYuanzhuLee**
## Summary
Adds a Triton kernel for `acos_`. In-place version of acos. Computes the arccosine (in radians) of each element of `input` and writes the result back to the input tensor.

## Testing
- Validated against reference on device via `to_reference(inp, True)`
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_acos_.py --level core` (NVIDIA H20)

### acos_ (in-place)

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup | TFLOPS |
|-------|------|--------------------|-------------------|---------|--------|
| float16 | [1073741824] | 1.888320 | 1.887776 | 1.000 | 0.569 |
| float16 | [64, 64] | 0.005504 | 0.005600 | 0.983 | 0.001 |
| float16 | [4096, 4096] | 0.035392 | 0.035392 | 1.000 | 0.474 |
| float16 | [64, 512, 512] | 0.035456 | 0.035520 | 0.998 | 0.472 |
| float16 | [1024, 1024, 1024] | 1.887520 | 1.888096 | 1.000 | 0.569 |
| float32 | [1073741824] | 2.275744 | 2.220352 | 1.025 | 0.484 |
| float32 | [64, 64] | 0.005600 | 0.005664 | 0.989 | 0.001 |
| float32 | [4096, 4096] | 0.039456 | 0.039584 | 0.997 | 0.424 |
| float32 | [64, 512, 512] | 0.039328 | 0.039456 | 0.997 | 0.425 |
| float32 | [1024, 1024, 1024] | 2.282032 | 2.246688 | 1.016 | 0.478 |
| bfloat16 | [1073741824] | 1.950048 | 1.950144 | 1.000 | 0.551 |
| bfloat16 | [64, 64] | 0.005408 | 0.005568 | 0.971 | 0.001 |
| bfloat16 | [4096, 4096] | 0.036416 | 0.036576 | 0.996 | 0.459 |
| bfloat16 | [64, 512, 512] | 0.036480 | 0.036576 | 0.997 | 0.459 |
| bfloat16 | [1024, 1024, 1024] | 1.950080 | 1.950016 | 1.000 | 0.551 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| acos_ (in-place) | **0.998** |

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20) | PASS (15 cases) | 0.998 | Primary |
| Tianshu | FAIL | 1.025 | AssertionError: Scalars are not close! |
| Muxi | PASS | 1.004 | — |
| Ascend | FAIL | — | CompilationError: at 2:11: NameError("Cannot access global v |
| Hygon | PASS | — | return lambda *args, **kwargs: self.run(grid=grid, wa |

## Files Changed
- `src/flag_gems/ops/acos_.py`: Triton kernel implementation
- `tests/test_acos_.py`: Accuracy test
- `benchmark/test_acos_.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.3)

### CODE REVIEW

**Review by Dongxu-H — COMMENTED** [review]
## 💡 审查建议

建议:
- ⚠️ 建议在flag_gems/__init__.py注册

**Review by Dongxu-H — COMMENTED** [review]
## 💡 审查建议

建议:
- ⚠️ 建议在flag_gems/__init__.py注册

**Review by Dongxu-H — COMMENTED** [review]
## 💡 审查建议

建议:
- ⚠️ 建议在flag_gems/__init__.py注册

**Comment by Dongxu-H** [comment]
## Code Review for acos_ Operator

### Overall Assessment ✅
Clean implementation of the in-place arccosine operation. The code follows the established patterns in the codebase. A few suggestions for improvement:

### Specific Comments

**1. Numerical Precision** (src/flag_gems/ops/acos_.py:28)
```python
return tl_extra_shim.acos(x.to(tl.float32))
```

Good practice converting to float32 before the computation! This ensures numerical precision. However, consider:
- What happens to the dtype of the output? Is it always float32 or does it get promoted back to the original dtype?
- For float16 input, converting to float32 is correct. For float64 input, this might lose precision. Consider:
```python
# Preserve float64 precision if input is float64
target_dtype = tl.float64 if x.dtype == tl.float64 else tl.float32
return tl_extra_shim.acos(x.to(target_dtype))
```

**2. In-Place Semantics** (src/flag_gems/ops/acos_.py:31-33)
```python
def acos_(A):
    logger.debug("GEMS ACOS_")
    acos_kernel(A, out0=A)
    return A
```

The in-place operation is correctly implemented. However:
- **Critical**: What happens to the dtype when the input is an integer type? The `promotion_methods=[(0, "INT_TO_FLOAT")]` suggests it converts int to float, but this changes the dtype in-place which may not be supported by all tensor types.
- Consider adding a dtype check or documenting the behavior for integer inputs.

**3. Missing Validation** 
Consider adding:
- Check that the input is a floating-point type (or document that integer inputs will be converted)
- Check that input values are in the valid range [-1, 1] for arccosine, or document that NaN will be produced for out-of-range values

**4. Test Coverage** (tests/test_acos_.py)

The test coverage is minimal. Consider adding:
- **Edge cases**: Values at -1, 0, 1 (boundary values)
- **Out-of-range values**: Values outside [-1, 1] should produce NaN
- **Integer input tests**: If supported, test dtype conversion
- **Different shapes**: Empty tensors, scalars
- **Gradient tests**: If this operator supports autograd

Example:
```python
def test_acos_boundary_values():
    # Test exact boundary values
    inp = torch.tensor([-1.0, 0.0, 1.0], dtype=torch.float32, device=flag_gems.device)
    # Should produce pi, pi/2, 0 respectively
    pass

def test_acos_out_of_range():
    # Values outside [-1, 1] should produce NaN
    inp = torch.tensor([-2.0, 2.0], dtype=torch.float32, device=flag_gems.device)
    # ...
```

**5. Code Organization**
The implementation is split across two files (acos and acos_). Consider whether they should share a common kernel or utilities to reduce code duplication.

### Minor Suggestions
- Add a docstring explaining the in-place semantics and any dtype behavior
- Consider whether to raise an error for integer inputs vs. auto-conversion

### Backend Specific (iluvatar)
The iluvatar backend implementation is identical to the main implementation. Consider whether code sharing is possible.

Overall: Good implementation, but would benefit from more comprehensive test coverage, especially edge cases and dtype handling. ✅

**Comment by Caeruleann** [comment]
Please fix failing checks.

File: src/flag_gems/runtime/backend/_iluvatar/ops/acos_.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+logger = logging.getLogger(__name__)
+
+
+@pointwise_dynamic(promotion_methods=[(0, "INT_TO_FLOAT")])
+@triton.jit()
+def acos_kernel(x):
+    return tl_extra_shim.acos(x.to(tl.float32))
+
+
+def acos_(A):
```
Comment (Caeruleann) [inline src/flag_gems/runtime/backend/_iluvatar/ops/acos_.py]: The Iluvatar override only differs from the main implementation in the logger string. Could we remove this override and let Iluvatar fall back to the main implementation automatically?

Comment (kkkwb) [reply src/flag_gems/runtime/backend/_iluvatar/ops/acos_.py]: Done — removed the Iluvatar `acos_` override so it falls back to the main implementation automatically. Commit 3f839fad3c72e10392a26258fff9f5ecd8a676ea.

**Review by Caeruleann — APPROVED** [review]
